res/controllers/Denon-MC6000MK2-scripts.js

```diff
@@ -1212,22 +1162,28 @@ DenonMC6000MK2.getEfxUnitByGroup = function(group) {
     return efxUnit;
 };
 
-DenonMC6000MK2.EfxUnit = function(side, group, paramGroup) {
+DenonMC6000MK2.EfxUnit = function(side, unit) {
+    //[EffectRack1_EffectUnit1_Effect1] "meta" / "enabled"
     this.side = side;
-    this.group = group;
-    DenonMC6000MK2.efxUnitsByGroup[group] = this;
-    this.paramGroup = paramGroup;
-    DenonMC6000MK2.efxUnitsByGroup[paramGroup] = this;
-    this.wetLoop = false;
-    this.wetLoopSlip = false;
-    this.echoLoopParams = false;
-    this.echoLoopDelayBeats = 1.0; // 1 beat echo
+    this.group = "[" + DenonMC6000MK2.EFX_RACK + "_" + unit + "]";
+    DenonMC6000MK2.efxUnitsByGroup[this.group] = this;
+    //this.wetLoop = false;
+    //this.wetLoopSlip = false;
+    //this.echoLoopParams = false;
+    //this.echoLoopDelayBeats = 1.0; // 1 beat echo
+    this.paramGroups = [];
     this.params = [];
-    this.params[1] = new DenonMC6000MK2.EfxParam(this);
-    this.params[2] = new DenonMC6000MK2.EfxParam(this);
-    this.params[3] = new DenonMC6000MK2.EfxParam(this, this.group, "mix");
-    this.dryWetMixParam = this.params[3];
-    this.dryWetMixParam.adjustParam(0.5); // mix (50% dry / 50% wet)
+    this.paramGroups[1] = "[" + DenonMC6000MK2.EFX_RACK + "_" + unit + "_Effect1]";
+    DenonMC6000MK2.efxUnitsByGroup[this.paramGroups[1]] = this;
+    this.params[1] = new DenonMC6000MK2.EfxParam(this, this.paramGroups[1], "meta");
```

Comment:
Use the [EffectUnit object from the Components library](https://mixxx.org/wiki/doku.php/components_js#effectunit). It behaves the same way by default and allows the user to control the parameters of each effect.